### ISSUE TITLE
Support for background colors

### ISSUE THREAD

**Issue by wooorm** [shikijs/shiki#318]
Hey!

I see that there’s internal support for background colors: 

https://github.com/shikijs/shiki/blob/747571082367aff6f1445318bad6b698df330e87/packages/shiki/src/stackElementMetadata.ts#L100

…and that certain themes use background colors (e.g., GH). However, it doesn’t seem like they are being retrieved when creating tokens (): 

https://github.com/shikijs/shiki/blob/747571082367aff6f1445318bad6b698df330e87/packages/shiki/src/themedTokenizer.ts#L135-L137

…Is there a reason they are not supported? Is it possible to support background?

---

Finally, a question, but not a blocker to me, the GH theme also includes:

```json
    {
      "scope": "carriage-return",
      "settings": {
        "fontStyle": "italic underline",
        "background": "#ff7b72",
        "foreground": "#f0f6fc",
        "content": "^M"
      }
    }
```

Is it correct that shiki does not support `content` and combined `fontStyle`s either, as it doesn’t fit in the binary format (https://github.com/shikijs/shiki/blob/747571082367aff6f1445318bad6b698df330e87/packages/shiki/src/stackElementMetadata.ts)

**Comment by orta** [comment]
I'd probably call this an oversight on the shiki renderer side for the main bit (and I have no clue about the latter :D )

**Comment by wooorm** [comment]
This issue was from when I started working on GHs highlighting, which in the end turned into its own project that inlines some of the stuff that `shiki` does, and this code in particular is inspired by `vscode-textmate` if I recall correctly.

I am personally no longer waiting on this issue to be solved, as I don‘t use Shiki myself, but I imagine others would be interested in solving it. In which case:
- Here’s the similar code with fixes that does include backgrounds: https://github.com/wooorm/starry-night/blob/main/lib/parse.js.
- Here’s the `vscode-textmate` reference now: https://github.com/microsoft/vscode-textmate/blob/main/src/encodedTokenAttributes.ts.

**Comment by octref** [comment]
Sorry for the late response. I think I omitted background support probably because I was using the `editor.background` from a theme, and I didn't find any theme that wants to highlight single words with different background. What theme does this?

**Comment by octref** [comment]
To me, I think the only case where a different background can be helpful is highlighted lines.

**Comment by antfu** [comment]
While it's not hard to support bg color for special tokens, but it seems VS Code somehow ignores the background on rendering as well - so in the sprit of aligning with VS Code, I guess we could consider this a won't fix at this moment, until something changed on VS Code side:

![Uploading image.png…]()
